std/conv.d

```diff
@@ -3785,6 +3788,16 @@ if (isInputRange!Source && isSomeChar!(ElementType!Source) && !is(Source == enum
         assertThrown!ConvException(parse!double(s));
 }
 
+@safe unittest // https://issues.dlang.org/show_bug.cgi?id=22637
+{
+    import std.exception : assertThrown, assertNotThrown;
+    auto src = "9991232549867999698999493543521458974414359998784641646846435132132543645435456345634541999999999999999"
+    ~ "9999999943321231321311999231345312413646846354354354399999934153465464654646464654134135354199999999996515734999"
+    ~ "9999999320135273486741354354731567431324134999999999999999999999999999999999999999999999135411.9";
+    assertThrown!ConvException(parse!double(src));
+    assertNotThrown!ConvException(parse!real(src));
```

Comment:
This assertion assumes `real` precision is higher than double's - which isn't the case for many architectures.

Reply:
Would adding `version(X86_64)` suffice, or should I remove the test case for `real` alltogether?

Reply:
You could use `static if (real.max_10_exp > 310)`

Reply:
I used `static if` as proposed by dkorpel